tools/iree-opt-main.cc

```diff
@@ -24,6 +24,16 @@ int main(int argc, char **argv) {
   mlir::iree_compiler::registerAllPasses();
   mlir::iree_compiler::registerHALTargetBackends();
 
+  // Also register the transform interpreter pass so that iree-opt can run
+  // transform dialect IR without resorting to a separate file.
+  // Resorting to separate files is a convenience for iree-compile to be able to
+  // use the transform dialect without requiring special plumbing.
+  // Still the preferred mode of execution should be to transport the relevant
+  // piece of transform IR in the right location, for each piece of code we
+  // want to transform for.
+  mlir::linalg::transform::registerTransformDialectInterpreterPass();
```

Comment:
My understanding is that the transform dialect is supposed to remain optional (documented in our current OKRs). How is this in keeping with that goal?

Reply:
But the dialect is in core. I think this is OK.

Reply:
Registration does not imply that it is mandatory to use it in production.
Optionality is builtin the whole process and even the whole MLIR codegen, pick and chose the best tool at hand for the job.